crates/topos-sequencer-subnet-runtime/src/proxy.rs

```diff
@@ -122,11 +122,17 @@ impl SubnetRuntimeProxy {
                                     "Source head certificate id received {:?}",
                                     certificate_and_position
                                 );
-                                // If the position is not provided, it should start form -1, so that first fetched is subnet genesis block
+                                // If tce source head position is provided, continue syncing from it
+                                // If `start_block` sequencer parameter is provided and tce source head is missing,
+                                // we should start syncing from that block instead of genesis
+                                // If both tce source head position and start_block parameter are not provided,
+                                // sync should start form -1, so that first fetched is subnet genesis block
+                                let default_block_sync: i128 =
```

Comment:
`i128`? Why?

Reply:
Because I can not indicate unknown (default starting) block with `u64`

Reply:
@dvdplm Block number is always positive u64. Last block number that sequencer previously retrieved from subnet node could be 7894 or could be 0 (we have genesis block), and then next block that should be read from subnet node is `latest_acquired_subnet_block_number` + 1. If we have a new subnet blockchain and we have no information about it we should read first genesis block 0. To indicate that we say that `latest_acquired_subnet_block_number` is invalid (value -1) so we can read genesis block as `latest_acquired_subnet_block_number`+1

Reply:
I see. Not super elegant. :/

Reply:
What would you do as alternative? I don't like the idea of introducing another Option to mark  if we have initial info about genesis block and handling everywhere through both cases for block number

Reply:
Good question. I'll have a think and see what I come up with.